src/server/pegasus_server_impl.cpp

```diff
@@ -1496,62 +1498,67 @@ ::dsn::error_code pegasus_server_impl::start(int argc, char **argv)
     //          3, restore_dir is exist
     //
     bool db_exist = true;
-    auto path = ::dsn::utils::filesystem::path_combine(data_dir(), "rdb");
-    if (::dsn::utils::filesystem::path_exists(path)) {
+    auto rdb_path = ::dsn::utils::filesystem::path_combine(data_dir(), "rdb");
```

Comment:
L1500 - L1559 have too many condiftions, doing following checks:
- db already exist
- db not exist (1. normal 2. duplication 3. restore)

Could you please refactor it?

Reply:
`refactor` is not this pr, which may introduce some other bug

Reply:
okay, expecting your further refactor pull request